integration/tests/basic/evm/test_solana_interoperability.py

```diff
@@ -56,6 +69,49 @@ def test_counter(self, call_solana_caller):
         instruction_tx = call_solana_caller.functions.execute(lamports, serialized).build_transaction(tx)
         resp = self.web3_client.send_transaction(sender, instruction_tx)
         assert resp["status"] == 1
+        event_logs = call_solana_caller.events.LogBytes().process_receipt(resp)
+        assert int.from_bytes(event_logs[0].args.value, byteorder="little") == next(get_counter_value)
```

Comment:
I checked again and it should works I think and start from 0 each new run 
because for saving data we use an account created by self.create_resource(call_solana_caller, "1", self.accounts[0], COUNTER_ID) and  self.accounts[0] is new each time

Reply:
Yes, you are right, it starts with 0 and incrementing after `execute` call and the counter is new for each new execution as you mentioned (now I understand why 🙂 ).